pkg/nfd-master/nfd-master.go

```diff
@@ -883,13 +914,24 @@ func (m *nfdMaster) refreshNodeFeatures(cli k8sclient.Interface, node *corev1.No
 	maps.Copy(extendedResources, crExtendedResources)
 	extendedResources = m.filterExtendedResources(features, extendedResources)
 
+	if m.config.Restrictions.DisableExtendedResources {
+		klog.InfoS("maximum number taints per CR restriction is applied, skipping")
+		extendedResources = map[string]string{}
+	}
+
 	// Annotations
 	annotations := m.filterFeatureAnnotations(crAnnotations)
 
 	// Taints
 	var taints []corev1.Taint
 	if m.config.EnableTaints {
-		taints = filterTaints(crTaints)
+		taints = m.filterTaints(crTaints)
+	}
+
+	// If we deny node feature labels, we'll empty the labels variable
+	if m.config.Restrictions.DenyNodeFeatureLabels {
```

Comment:
This setting should only disable labels from NodeFeature objects. Not all labels. And more specifically, it should (I think) disable labels from __3rd party__ NodeFeature objects. That is, leave the built-in labels from nfd-worker intact but deny 3rd party extensions from creating node labels directly. Built-in labels can be disabled separately with nfd-worker config if desired. WDYT?

So, this should be done in `nfdAPIUpdateOneNode()` where the NF objects are handled.